### PR TITLE
Feat/add app open support

### PR DESCRIPTION
**Pull request by AviNLeung**
Asana Ticket URL: https://app.asana.com/0/1203775898402819/1203941004305016

I am submitting the Android PR request before the IOS one as I did with a previous pull request, as it seems that this is a bigger PR, so wanted to get this in ASAP. 

Testing Notes: 
1. Make sure to add your SDK key to the android manifest file. 
2. Put the app in test mode and run the App Open Ads under the MAX menu. 

Screenshot_20230210-165040
Screenshot_20230210-165048
Screenshot_20230213-131343

### CODE REVIEW

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/AppOpenAdActivity.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        extends BaseAdActivity
+        implements MaxAdListener, MaxAdRevenueListener
+{
+    private MaxAppOpenAd appOpenAd;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState)
+    {
+        super.onCreate( savedInstanceState );
+        setContentView( R.layout.activity_app_open_ad_ );
```
Comment (Lorenzo45) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/AppOpenAdActivity.java:32]: nit: Remove trailing `_` at the end of the file name (need to update xml file name too)

File: AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/AppOpenAdActivity.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * <p>
+ * Created by avileung on 2023-02-10.
+ */
+class AppOpenAdActivity : BaseAdActivity(),
+        MaxAdListener, MaxAdRevenueListener {
+    private lateinit var appOpenAd: MaxAppOpenAd
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_app_open_ad_)
```
Comment (Lorenzo45) [inline AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/AppOpenAdActivity.kt:27]: nit: Remove trailing `_` at the end of the file name (need to update xml file name too)

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/AppOpenAdActivity.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    {
+        logCallback();
+    }
+
+    @Override
+    public void onAdDisplayFailed(final MaxAd ad, final MaxError error)
+    {
+        logCallback();
+
+        // appOpen ad failed to display. We recommend loading the next ad.
```
Comment (Lorenzo45) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/AppOpenAdActivity.java:74]: nit: Fix comment
```suggestion
        // App Open ad failed to display. We recommend loading the next ad.
```

File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/AppOpenAdActivity.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    @Override
+    public void onAdClicked(final MaxAd ad) { logCallback(); }
+
+    @Override
+    public void onAdHidden(final MaxAd ad)
+    {
+        logCallback();
+
+        // appOpen Ad is hidden. Pre-load the next ad
```
Comment (Lorenzo45) [inline AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/ads/max/AppOpenAdActivity.java:89]: nit: Fix comment
```suggestion
        // App Open ad is hidden. Pre-load the next ad
```

File: AppLovin MAX Demo App - Java/app/src/main/res/layout/activity_app_open_ad_.xml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        android:id="@+id/callbacks_recycler_view"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        android:padding="10dp"
+        app:layout_constraintBottom_toTopOf="@id/show_ad_button"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent" >
+
+    </androidx.recyclerview.widget.RecyclerView>
```
Comment (Lorenzo45) [inline AppLovin MAX Demo App - Java/app/src/main/res/layout/activity_app_open_ad_.xml:19]: nit: You can combine this into one XML tag (you only need the start/end tag if you have tags inside)
```suggestion
        app:layout_constraintTop_toTopOf="parent" />
```

File: AppLovin MAX Demo App - Kotlin/app/src/main/res/layout/activity_app_open_ad_.xml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/callbacksRecyclerView"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"
+        app:layout_constraintBottom_toTopOf="@id/showAdButton"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        android:padding="10dp"/>
```
Comment (Lorenzo45) [inline AppLovin MAX Demo App - Kotlin/app/src/main/res/layout/activity_app_open_ad_.xml:17]: nit: You can actually use `cmd + opt + L` on these files to clean up these attributes a bit

**Review by Lorenzo45 — APPROVED** [review]
Looks good 👍 Just a bit of cleanup

**Comment by AviNLeung** [comment]
Changes Made

**Review by Lorenzo45 — APPROVED** [review]

**Review by thomasmso — APPROVED** [review]

**Comment by candratop3** [comment]
it will be changed in 11.7.2 ???

**Comment by Lorenzo45** [comment]
@candratop3 no changes to app open ads planned for the next SDK. This PR just adds app open ads to our demo apps (already supported in SDK 11.7.1)

**Pull request merged**